component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,323 @@
+                                "status string);\n" +
+                                "@sink(type='log')\n" +
+                                "define stream FooStream (filepath string, filename string, status string); \n" +
+                                "from FileListenerStream\n" +
+                                "select *\n" +
+                                "insert into FooStream;",
+
+                        description = "" +
+                                "Under above configuration,  An event is triggered if any file under the given" +
+                                "URI gets created, modified or deleted in the execution time. " +
+                                "An event is created with the filepath, filename " +
+                                "and status of the file.Then that will be received by the FooStream..\n"
+                ),
+                @Example(
+                        syntax = "" +
+                                "@source(type='fileeventlistener',uri='file://abc/xyz', monitoring.interval='200')\n" +
+                                "define stream FileListenerStream (filepath string, filename string, " +
+                                "status string);\n" +
+                                "@sink(type='log')\n" +
+                                "define stream FooStream (filepath string, filename string, status string);\n" +
+                                "from FileListenerStream\n" +
+                                "select *\n" +
+                                "insert into FooStream;",
+
+                        description = "" +
+                                "Under above configuration, An event is triggered if any file under the given URI " +
+                                "gets created, modified or deleted in the execution time. " +
+                                "An event is created with the filepath, filename and " +
+                                "status of the file. Then that will be received by the FooStream. If there " +
+                                "are any changes a new event will be generated in every 200 milliseconds.\n"
+                ),
+        }
+)
+
+public class FileHandler extends Source<FileHandler.FileHandlerState> {
+    private static final Logger log = Logger.getLogger(FileHandler.class);
+    private static final String EMPTY_STRING = "";
+    private SourceEventListener sourceEventListener;
+    private long monitoringInterval = 100;
+    private String listeningDirUri;
+    private FileAlterationMonitor monitor;
+    private Map<String, Long> fileObjectMap = new ConcurrentHashMap<>();
+    private static final String CURRENT_MAP_KEY = "current.map.key";
+    private List<String> fileObjectList;
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public StateFactory<FileHandlerState> init(SourceEventListener sourceEventListener,
```

Comment:
Shall we try to arrange this into 3 lines? rather than having across 5

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/216bbff54ceeaef2da8cba05aab1111e245f6e91